server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -257,6 +274,10 @@ public Sink getSink(long timestamp)
                                           @Override
                                           public QueryRunner<T> apply(FireHydrant input)
                                           {
+                                            if (input == null || input.getSegment() == null) {
```

Comment:
can we add a comment to indicate in which situations either input or input.getSegment will be null?